app.js

```diff
@@ -399,3 +399,25 @@ form.addEventListener('submit', (event) => {
     error.className = 'error';
   }
 });
+
+/* Preserve to local storage */
+
+if (localStorage.getItem('formData')) {
+  const formData = JSON.parse(localStorage.getItem('formData'));
+  document.getElementById('fullname').value = formData.name;
+  document.getElementById('mail').value = formData.email;
+  document.getElementById('message').value = formData.message;
+}
```

Comment:
- [ ] kindly consider making sure that as I type a single letter in any input it's immediately saved in the local storage so that After filling out the form, and not submitting it, the entered values should be saved in the browser and when visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page. because currently the data are only being saved when the form is submitted